### ISSUE TITLE
WIN32: Data Corruption on minigw/gcc 

### ISSUE THREAD

**Issue by sithlord48** [sithlord48/ff7tk#8]
for some reason __attribute__((packed)) is not working as expected on this platform.

**Comment by sithlord48** [comment]
Vegetta has researched and fixed this in commit : https://github.com/sithlord48/ff7tk/commit/4911ed4ed27551d9ceedf2442e38d0d4ea755a10
Posted in case others have a similar issue so they can also fix it.

**Comment by myst6re** [comment]
I use a strange code to compact my structs for msvc and mingw:

```C++
#ifdef _MSC_VER
#	define PACK(structure)			\
		__pragma(pack(push, 1))		\
		structure					\
		__pragma(pack(pop))
#else
#	define PACK(structure) structure Q_PACKED
#endif
```

To use it, you need to wrap your structure:

```C++
PACK(
struct MyStruct
{
	int foo, bar;
}
);
```

That's a way.

**Comment by sithlord48** [comment]
we have fixed this by adding 
QMAKE_CXXFLAGS += -mno-ms-bitfields 
to the pro file if the compiler is GCC. seams to have no affect on linux but GCC on windows with mingw is now using the -mms-bitfields method of padding for aligned structs

we have yet to mess with using msvc since gcc is basicly everywhere.

**Comment by vegetass4** [comment]
`QMAKE_CXXFLAGS += -mno-ms-bitfields` prevent on ALL gcc compiler (including MINGW). It no matter if you are on linux or windows (using FREE MINGW) this gcc flag prevent gcc use ms-bitfields in all scenarios.

We all know about MS proprietary licenses over the years. If a fact that Microsoft model is against GNU development and LGP licenses. For all this reasons patch all gnu code up to make it work on buggy MSVS is nowhere near our priority list. 

Thanks for your idea Myst, is actually very interesting.

**Comment by vegetass4** [comment]
Sithlord implemented Myst fix at  bb9eb988178097ab2b4c28ba97aa015509d2ba8b

I can confirm is working ok for both mingw & msvc.

Great job team!
